core/src/test/java/com/linecorp/armeria/client/ConnectionPoolCollectingMetricTest.java

```diff
@@ -0,0 +1,120 @@
+            sb.service("/", (ctx, req) -> HttpResponse.of(OK));
+        }
+
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
+    };
+
+    @RegisterExtension
+    static ServerExtension server2 = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.idleTimeoutMillis(0);
+            sb.requestTimeoutMillis(0);
+            sb.service("/", (ctx, req) -> HttpResponse.of(OK));
+        }
+
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
+    };
+
+    @BeforeEach
+    void setUp() {
+        connectionPoolListener = ConnectionPoolListener.metricCollecting(registry);
+    }
+
+    @EnumSource(value = SessionProtocol.class, names = "PROXY", mode = EnumSource.Mode.EXCLUDE)
+    @ParameterizedTest
+    void test(SessionProtocol protocol) throws Exception {
+        try (ClientFactory factory = ClientFactory.builder()
+                                                  .connectionPoolListener(connectionPoolListener)
+                                                  .idleTimeoutMillis(0)
+                                                  .maxConnectionAgeMillis(1000)
+                                                  .tlsNoVerify()
+                                                  .build()) {
+            final WebClient client = WebClient.builder(server.uri(protocol))
+                                              .factory(factory)
+                                              .responseTimeoutMillis(0)
+                                              .build();
+
+            assertThat(client.get("/").aggregate().join().status()).isEqualTo(OK);
+
+            assertThat(MoreMeters.measureAll(registry))
+                    .containsEntry("armeria.client.connections#count{state=open}", 1.0)
+                    .containsEntry("armeria.client.connections#count{state=close}", 0.0);
+
+            final WebClient client2 = WebClient.builder(server2.uri(protocol))
+                                               .factory(factory)
+                                               .responseTimeoutMillis(0)
+                                               .build();
+
+            assertThat(client2.get("/").aggregate().join().status()).isEqualTo(OK);
+
+            assertThat(MoreMeters.measureAll(registry))
+                    .containsEntry("armeria.client.connections#count{state=open}", 2.0)
+                    .containsEntry("armeria.client.connections#count{state=close}", 0.0);
+
+            await().untilAsserted(() -> {
+                assertThat(MoreMeters.measureAll(registry))
+                        .containsEntry("armeria.client.connections#count{state=open}", 2.0)
+                        .containsEntry("armeria.client.connections#count{state=close}", 2.0);
+            });
+        }
```

Comment:
Alternatively, we can close the `ClientFactory` and then await for the meterRegistry to be updated.

When testing, this reduced testing time 50s -> 10s
```suggestion
        }
        await().untilAsserted(() -> {
                assertThat(MoreMeters.measureAll(registry))
                        .containsEntry("armeria.client.connections#count{state=open}", 2.0)
                        .containsEntry("armeria.client.connections#count{state=close}", 2.0);
            });
```